hbase-server/src/main/java/org/apache/hadoop/hbase/replication/HBaseReplicationEndpoint.java

```diff
@@ -288,39 +258,26 @@ public boolean isAborted() {
     return false;
   }
 
-  /**
-   * Get the connection to peer cluster
-   * @return connection to peer cluster
-   * @throws IOException If anything goes wrong connecting
-   */
-  private synchronized AsyncClusterConnection getPeerConnection() throws IOException {
-    if (peerConnection == null) {
-      Configuration conf = ctx.getConfiguration();
-      peerConnection = ClusterConnectionFactory.createAsyncClusterConnection(conf, null,
-          UserProvider.instantiate(conf).getCurrent());
-    }
-    return peerConnection;
-  }
-
   /**
    * Get the list of all the servers that are responsible for replication sink
    * from the specified peer master
    * @return list of server addresses or an empty list if the slave is unavailable
```

Comment:
Is it true that an empty list would be returned if we can't reach slave master? Looks like we either throw an IOException or a RemoteException.

Reply:
Yes, I ignored changing the comment. 
Fixed it.